packages/phone-number-privacy/combiner/test/signing/bls-signature.test.ts

```diff
@@ -147,12 +147,73 @@ describe(`BLS service computes signature`, () => {
 
     const blsCryptoClient = new BLSCryptographyClient()
     await signatures.forEach(async (signature) => {
-      await blsCryptoClient.addSignature(signature, blindedMsg)
+      await blsCryptoClient.addSignature(signature)
     })
     try {
-      await blsCryptoClient.combinePartialBlindedSignatures()
+      await blsCryptoClient.combinePartialBlindedSignatures(blindedMsg)
     } catch (e) {
       expect(e.message.includes('Not enough not enough partial signatures'))
     }
   })
+  it('throws error if signature is invalid, but can recover from failure with sufficient signatures', async () => {
+    const signatures: ServicePartialSignature[] = [
+      {
+        url: 'url1',
+        signature:
+          'MAAAAAAAAABMnIduYMm1JmaOWWgybOTc6rB7+eunT9h21RL2oTId3KSOH1OjAglgOpI6cjizLQEAAAAA',
+      },
+      {
+        url: 'url2',
+        signature: 'X',
+      },
+      {
+        url: 'url3',
+        signature:
+          'MAAAAAAAAAD60iBC0rpJd9A+FjDzVix/xjdD5Rq8+euqX/pTJuwzooTXu/9+KBztQruAAAYWtAACAAAA',
+      },
+      {
+        url: 'url4',
+        signature:
+          'MAAAAAAAAADNYzSf29At1wxuqPXcvNYsYObrxZTmPXgg0KBH+BZm1lLczhv8NpedtgkPjX+GvwADAAAA',
+      },
+    ]
+
+    const message = Buffer.from('hello world')
+    const userSeed = new Uint8Array(32)
+    for (let i = 0; i < userSeed.length - 1; i++) {
+      userSeed[i] = i
+    }
+
+    const blindedMsgResult = threshold_bls.blind(message, userSeed)
+    const blindedMsg = Buffer.from(blindedMsgResult.message).toString('base64')
+
+    const blsCryptoClient = new BLSCryptographyClient()
+    // Add sigs one-by-one and verify intermediary states
+    await blsCryptoClient.addSignature(signatures[0])
+    expect(blsCryptoClient.hasSufficientSignatures()).toBeFalsy()
+    await blsCryptoClient.addSignature(signatures[1])
+    expect(blsCryptoClient.hasSufficientSignatures()).toBeFalsy()
+    await blsCryptoClient.addSignature(signatures[2])
+    expect(blsCryptoClient.hasSufficientSignatures()).toBeTruthy()
+    // Should fail since 1/3 sigs are invalid
+    try {
+      await blsCryptoClient.combinePartialBlindedSignatures(blindedMsg)
+    } catch (e) {
+      expect(e.message.includes('Not enough not enough partial signatures'))
```

Comment:
Shouldn't this be 'CELO_ODIS_ERR_06 SIG_ERR BLS partial signature verification Failure'?

Reply:
I also wonder if maybe this check would pass even if no error was thrown because the expect statement is in the catch block

Reply:
And actually this string looks suspicious because 'not enough' is repeated

Reply:
Good eyes! We needed to change it to `expect(e.message.includes('Not enough partial signatures')).toBeTruthy()`